### ISSUE TITLE
Multiple @HostBinding decorators do not work in 2.3.0

### ISSUE THREAD

**Issue by krassx** [angular/angular#13303]
**I'm submitting a ...**  (check one with "x")
```
[ x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Multiple @HostBinding decorators re not handled properly. Only last one is used.

**Expected behavior**
All decorators must be used

**Minimal reproduction of the problem with instructions**
Create a component and set multiple HostBindings to single field, like:
```
@Input()
@HostBinding("style.width.px")
@HostBinding("style.height.px")
@HostBinding("style.min-width.px")
@HostBinding("style.min-height.px")
public size: number = 30;
```
Only last decorator is used when app is compiled, both in JIT and AoT.

**What is the motivation / use case for changing the behavior?**
Does not work as expected.

**Please tell us about your environment:**
Ubuntu 16.04

* **Angular version:** 2.0.X
2.3.0

* **Browser:** [all]
 
* **Language:** [TypeScript 2.0.10]

* **Node (for AoT issues):** `node --version` =   4.2.6

**Comment by krassx** [comment]
Btw, it works in 2.2.4.

**Comment by DzmitryShylovich** [comment]
Are u sure? Input and HostBinding don't work on the same prop since rc https://github.com/angular/angular/issues/10499

**Comment by laco0416** [comment]
plunkr repros are needed

**Comment by DzmitryShylovich** [comment]
Repro http://plnkr.co/edit/IvSU6eI99ZB4N55yswko?p=preview

**Comment by vicb** [comment]
@DzmitryShylovich is it working / a bug / a duplicate of #10499 ?

**Comment by DzmitryShylovich** [comment]
@vicb it's not a duplicate of #10499
> is it working / a bug

it depends on whether multiple decorators on the same property suppose to work. I don't know if it was designed like this :)

**Comment by vicb** [comment]
> it depends on whether multiple decorators on the same property suppose to work.

What do you mean ? Could you pleased expand ? What is the current behavior ?

**Comment by DzmitryShylovich** [comment]
@vicb given:
```
@HostBinding("style.width.px")
@HostBinding("style.height.px")
@HostBinding("style.min-width.px")
@HostBinding("style.min-height.px")
public size: number = 30;
```
only `@HostBinding("style.min-height.px")` does work.

**Comment by vicb** [comment]
Thanks much clearer, what happens if the config is in the decorator ? 

_There is no reason why we should not support this I guess_

**Comment by QuentinFchx** [comment]
It's also not working for `@HostListener` decorator.

**Comment by DzmitryShylovich** [comment]
I think it doesn't work for any decorator.

**Comment by BartlM** [comment]
Quite an easy workaround would be to declare a private member for each bound style property and fill their values in the setter of the orignal member.

`@HostBinding('style.width')`
`private cssWidth: string;`
`@HostBinding('style.min-width')`
`private cssMinWidth: string;`
`@HostBinding('style.max-width')`
`private cssMaxWidth: string;`

**Comment by krassx** [comment]
I do not fill comfortable with such workarounds as this will involve many refactorings, tests, etc.  In fact, what I was complaining about is a regression. As I noted above in 2.2.4 everything works perfectly.

I can live without combining Input and HostBinding (what is the reason to make them incompatible on single field then?). But I definitely want to have an opportunity to use multiple decorators for single field, otherwise there will be plenty of fields that are used just to assign same value for different styles, attrs, etc.

**Comment by BartlM** [comment]
I agree that a fix should be provided. I for sure can tell it worked with 2.0.0. Can`t tell about 2.2.4 because we upgraded from 2,0 to 2.3.

**Comment by vicb** [comment]
The issue has been introduced here: https://github.com/angular/angular/commit/f5c8e09#diff-269923e65df2d7a833af1d6468dce23bR78

**Comment by vicb** [comment]
fixed by #13474
